src/components/Reown/SendTransactionRequest.js

```diff
@@ -0,0 +1,492 @@
+import { COLORS } from '../../styles/themes';
+import NewHathorButton from '../NewHathorButton';
+import { WarnDisclaimer } from './WarnDisclaimer';
+import { DappContainer } from './NanoContract/DappContainer';
+import {
+  setSendTxStatusReady,
+  unregisteredTokensDownloadRequest,
+  reownReject,
+  sendTxRetry,
+  sendTxRetryDismiss
+} from '../../actions';
+import { REOWN_SEND_TX_STATUS, DEFAULT_TOKEN } from '../../constants';
+import FeedbackModal from '../FeedbackModal';
+import errorIcon from '../../assets/images/icErrorBig.png';
+import { FeedbackContent } from '../FeedbackContent';
+import Spinner from '../Spinner';
+import { DeclineModal } from './NanoContract/DeclineModal';
+import { useBackButtonHandler } from '../../hooks/useBackButtonHandler';
+
+const styles = StyleSheet.create({
+  wide: {
+    width: '100%',
+  },
+  wrapper: {
+    flex: 1,
+    paddingHorizontal: 16,
+    backgroundColor: COLORS.lowContrastDetail,
+  },
+  content: {
+    flex: 1,
+    rowGap: 24,
+    width: '100%',
+    paddingVertical: 16,
+  },
+  sectionTitle: {
+    fontSize: 16,
+    fontWeight: 'bold',
+    marginBottom: 16,
+  },
+  itemContainer: {
+    backgroundColor: COLORS.backgroundColor,
+    borderRadius: 12,
+    padding: 16,
+    marginBottom: 12,
+    shadowColor: COLORS.black,
+    shadowOffset: { width: 0, height: 2 },
+    shadowOpacity: 0.08,
+    shadowRadius: 4,
+    elevation: 2,
+  },
+  itemHeader: {
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    marginBottom: 12,
+    alignItems: 'center',
+  },
+  itemTitle: {
+    fontWeight: 'bold',
+    fontSize: 16,
+  },
+  valueText: {
+    fontSize: 16,
+    fontWeight: '600',
+  },
+  labelText: {
+    fontSize: 14,
+    color: 'hsla(0, 0%, 38%, 1)',
+    fontWeight: 'bold',
+    marginBottom: 4,
+  },
+  valueContainer: {
+    backgroundColor: COLORS.lowContrastDetail,
+    borderRadius: 8,
+    padding: 12,
+    marginBottom: 12,
+  },
+  monospace: {
+    fontFamily: 'Courier',
+    fontSize: 14,
+  },
+  copyButton: {
+    color: 'hsla(263, 100%, 64%, 1)',
+    marginTop: 4,
+  },
+  addressContainer: {
+    backgroundColor: COLORS.lowContrastDetail,
+    borderRadius: 8,
+    padding: 12,
+    marginBottom: 12,
+  },
+  actionContainer: {
+    flexDirection: 'column',
+    gap: 8,
+    paddingBottom: 48,
+  },
+  loadingContainer: {
+    alignItems: 'center',
+    justifyContent: 'center',
+    padding: 16,
+  },
+  loadingText: {
+    fontSize: 16,
+    marginTop: 16,
+    textAlign: 'center',
+  },
+  successContainer: {
+    alignItems: 'center',
+    justifyContent: 'center',
+    padding: 16,
+  },
+  successText: {
+    fontSize: 16,
+    marginBottom: 16,
+    textAlign: 'center',
+    color: 'green',
+  },
+  errorIcon: {
+    fontSize: 48,
+  },
+  feedbackModalIcon: {
+    width: 48,
+    height: 48,
+  },
+});
+
+export const SendTransactionRequest = ({ sendTransactionRequest, onAccept, onReject }) => {
```

Comment:
This component is really similar to `src/components/Reown/SendTransactionModal.js`. Why do you need both and if you need them, can you reuse the similar code?

Reply:
Sorry for that, refactored the SendTransactionModal to be the pre-modal just like the other RPC Requests

Also created a reusable component to reuse in those modals